operatorcourier/flatten.py

```diff
@@ -0,0 +1,151 @@
+
+    contains_csv = False
+    for item in os.listdir(os.path.join(base_dir, version_folder_name)):
+        version_folder_path = os.path.join(base_dir, version_folder_name)
+        item_path = os.path.join(version_folder_path, item)
+
+        if not os.path.isfile(item_path):
+            logger.warning('Ignoring %s as it is not a regular file.', item)
+
+        with open(item_path, 'r') as f:
+            file_content = f.read()
+
+        yaml_type = identify.get_operator_artifact_type(file_content)
+
+        if yaml_type == 'ClusterServiceVersion':
+            contains_csv = True
+            csv_paths.append(item_path)
+        elif yaml_type == 'CustomResourceDefinition':
+            try:
+                crd_name = safe_load(file_content)['metadata']['name']
+            except MarkedYAMLError:
+                msg = "Courier requires valid input YAML files"
+                logger.error(msg)
+                raise errors.OpCourierBadYaml(msg)
+            # create new CRD type entry if not found in dict
+            if crd_name not in crd_dict:
+                crd_dict[crd_name] = (version_folder_name, item_path)
+            # update the CRD type entry with the file with the newest version
+            elif semver.compare(version_folder_name, crd_dict[crd_name][0]) == 1:
+                crd_dict[crd_name] = (crd_dict[crd_name][0], item_path)
+
+    if not contains_csv:
+        msg = 'This version directory does not contain any valid CSV file.'
+        logger.error(msg)
+        raise errors.OpCourierBadBundle(msg)
+
+
+def get_package_file(base_dir: str, file_names_in_base_dir) -> (str, str):
+    packages = []  # [ (PACKAGE_FILE_PATH, PACKAGE_FILE_NAME) ]
+
+    # add package file to file_paths_to_copy
+    # only 1 package yaml file is expected in file_names
+    for file_name in file_names_in_base_dir:
+        file_path = os.path.join(base_dir, file_name)
+        with open(file_path, 'r') as f:
+            file_content = f.read()
+        if identify.get_operator_artifact_type(file_content) != 'Package':
+            logger.warning('Ignoring %s as it is not a valid package file.', file_name)
+        else:
+            packages.append((file_path, file_name))
+
+    n_packages = len(packages)
+    if n_packages != 1:
+        msg = f'The input source directory expects 1 valid package file, ' \
+            f'but got {n_packages}'
+        logger.error(msg)
+        raise errors.OpCourierBadBundle(msg)
+    else:
+        return packages[0][0], packages[0][1]
+
+
+# parse all CSVs and ensure those with same names are handled
+def create_csv_dict(csv_paths: list) -> dict:
+    csv_dict = {}  # { CSV_FILE_NAME => [ (v1, csv_path_1), ... , (vn, csv_path_n) ] }
+    for csv_path in csv_paths:
+        version, csv_file_name = csv_path.split('/')[-2:]
+        val = (version, csv_path)
+        if csv_file_name in csv_dict:  # duplicate CSVs found in different versions
```

Comment:
For this `if.. else` block there is a Pythonic way:

```suggestion
        csv_dict.setdefault(csv_file_name, []).append(val)
```

Read [setdefault](https://docs.python.org/3/library/stdtypes.html#dict.setdefault).